amethyst_animation/src/util.rs

```diff
@@ -0,0 +1,85 @@
+use amethyst_assets::Handle;
+use specs::{Entity, WriteStorage};
+
+use resources::{Animation, AnimationCommand, AnimationControl, ControlState, EndControl};
+
+/// Play a given animation on the given entity.
+///
+/// ## Parameters:
+///
+/// - `controls`: animation control storage in the world.
+/// - `animation`: handle to the animation to run
+/// - `entity`: entity to run the animation on. Must either have an `AnimationHierarchy` that
+///             matches the `Animation`, or only refer to a single node, else the animation will
+///             not be run.
+/// - `end`: action to perform when the animation has reached its end.
+pub fn play_animation(
+    controls: &mut WriteStorage<AnimationControl>,
+    animation: &Handle<Animation>,
+    entity: Entity,
+    end: EndControl,
+) {
+    if !match controls.get_mut(entity) {
```

Comment:
`if !match` is poorly readable.
Use single `match` block with guards or `if let Some` if appropriate.

Reply:
The problem is borrowing rules. I'd have to break it up into two `if let ..`